### ISSUE TITLE
Is there a more direct way to get PullRequest from Issue?

### ISSUE THREAD

**Issue by davidxia** [PyGithub/PyGithub#572]
Thanks for the awesome repo and work. It's made my life a lot easier.

I'm writing a script to comment on all open PRs created by me in order to remind people to review them. It turns out the most direct way to list all my open PRs is to use the issue search API endpoint. [See discussion here](http://stackoverflow.com/a/43807803/553994).

This works great and is better than my original [roundabout way](http://stackoverflow.com/questions/17412809/how-to-get-my-pull-requests-from-github-api/43807803#comment74624202_17438627).

Unfortunately, `Github.search_issues()` returns `Issue` and there's no easy way to get the associated `PullRequest` object. There's `Issue.pull_request`, but this returns an [`IssuePullRequest`](http://pygithub.readthedocs.io/en/latest/github_objects/IssuePullRequest.html#github.IssuePullRequest.IssuePullRequest) object which isn't helpful.

My current workaround is `Issue` -> `Repository` -> `PullRequest`.

```python
issue.repository.get_pull(issue.number)
```

Is there a better way to get the corresponding `PullRequest` from an `Issue`?

**Comment by mikeage** [comment]
I use pygithub for parsing webhooks, and as most of the notifications about a pull request come as an `issue_comment` notification, I have the same challenge you do; I haven't found anything better than `issue.repository.get_pull(issue.number)`. 

Not sure if that's the answer you're looking for, but at least I can confirm that you're not alone

**Comment by tmshn** [comment]
I added the methods for this!
:point_right: https://github.com/PyGithub/PyGithub/pull/630

**Comment by sfdye** [comment]
Yes, there is now. You can use the `type` parameter to specify only search against PR:
https://developer.github.com/v3/search/#parameters-3

Something like:

`g.search_issues(query="query", qualifiers={'type':'pr'})`

**Comment by sfdye** [comment]
Feel free to re-open if needed.

**Comment by tmshn** [comment]
Thanks @sfdye, but I'm afraid you are seeing wrong point (though I can't re-open this cz I'm not the author).

Even if you specify `{'type':'pr'}` on `search_issues`--well that's true it returns only pull requests, but it's represented in the `github.Issue.Issue` class. This is that kind of interface.

```python
from github import Github

g = Github()

for pr in g.search_issues(query='License', **{'repo':'PyGithub/PyGithub', 'type':'pr'}):
    print('{}\n  - type: {}\n  - URL: {}'.format(pr, type(pr), pr.html_url))
```
```
Issue(title="Implement License API", number=734)
  - type: <class 'github.Issue.Issue'>
  - URL: https://github.com/PyGithub/PyGithub/pull/734
Issue(title="Added missing parameters for repo creation", number=623)
  - type: <class 'github.Issue.Issue'>
  - URL: https://github.com/PyGithub/PyGithub/pull/623
Issue(title="Follow redirects in the case of a 301 status code", number=403)
  - type: <class 'github.Issue.Issue'>
  - URL: https://github.com/PyGithub/PyGithub/pull/403
```

The point is: we want convert `github.Issue.Issue` class instance which represents pull requests into `github.PullRequest.PullRequest` class instance.

**Comment by sfdye** [comment]
I understand that. Is there a specific use case to convert between the two types instead of retrieving the correct one in the first place?

**Comment by sfdye** [comment]
My bad, it seems your PR is still required. Will re-open.

**Comment by tmshn** [comment]
> Is there a specific use case to convert between the two types instead of retrieving the correct one in the first place?

[Issue→PR]
Yes, simply because Github provides no API to "search pull request" directly.

[PR→Issue]
Mmm, probably no, I think.

**Comment by sfdye** [comment]
Closed in #630 

@tmshn Thanks for your PR!
